src/unstract/sdk/tool/validator.py

```diff
@@ -70,10 +68,14 @@ def validate_pre_execution(
             dict[str, Any]: Settings JSON for a tool (filled with defaults)
         """
         input_file = Path(self.tool.get_input_file())
-        if not input_file.is_file():
-            self.tool.stream_error_and_exit(
-                f"Input file not found: {input_file}"
-            )
+        file_exists = (
+            self.tool.workflow_filestorage.exists(path=input_file)
+            if self.tool.workflow_filestorage
+            else input_file.is_file()
```

Comment:
Can we make use of the FS APIs to check the file instead of Pathlib? This will maintain complete abstractions on all file ops.
@gaya3-zipstack If so, we might need to add a method in SDK for this.

Reply:
else is for backward cmompatibility not for FS